figures/S7.supplemental_figure7/scripts/S7.r

```diff
@@ -0,0 +1,288 @@
+metadata_file_path_SHSY5Y <-file.path(
+    "..",
+    "..",
+    "..",
+    "data",
+    "SHSY5Y_preprocessed_sc_norm.parquet"
+)
+# load the metadata file via arrow
+metadata_file_SHSY5Y <- arrow::read_parquet(
+    metadata_file_path_SHSY5Y,
+    col_select = c(
+        'Metadata_cell_type',
+        'Metadata_Well'	,
+        'Metadata_number_of_singlecells',
+    )
+)
+head(metadata_file_SHSY5Y)
+
+
+
+
+
+## PBMC
+
+# add column to metadata file with the labeled_data_index
+metadata_file_PBMC$label <- index_file_PBMC$label
+# replace test, train, or validation with "pool" in label column
+metadata_file_PBMC$label <- gsub(
+    pattern = "test",
+    replacement = "pool",
+    x = metadata_file_PBMC$label
+)
+metadata_file_PBMC$label <- gsub(
+    pattern = "train",
+    replacement = "pool",
+    x = metadata_file_PBMC$label
+)
+metadata_file_PBMC$label <- gsub(
+    pattern = "val",
+    replacement = "pool",
+    x = metadata_file_PBMC$label
+)
+# aggreagate the metadata via Metadata_Well
+metadata_file_PBMC <- metadata_file_PBMC %>%
+    group_by(Metadata_Well, Metadata_cell_type, label) %>%
+    summarize(
+        n = n()
+    )
+head(metadata_file_PBMC)
+
+# SHSY5Y
+# add column to metadata file with the labeled_data_index
+metadata_file_SHSY5Y$label <- index_file_SHSY5Y$label
+# replace test, train, or validation with "pool" in label column
+metadata_file_SHSY5Y$label <- gsub(
+    pattern = "test",
+    replacement = "pool",
+    x = metadata_file_SHSY5Y$label
+)
+metadata_file_SHSY5Y$label <- gsub(
+    pattern = "train",
+    replacement = "pool",
+    x = metadata_file_SHSY5Y$label
+)
+metadata_file_SHSY5Y$label <- gsub(
+    pattern = "val",
+    replacement = "pool",
+    x = metadata_file_SHSY5Y$label
+)
+# aggreagate the metadata via Metadata_Well
+metadata_file_SHSY5Y <- metadata_file_SHSY5Y %>%
+    group_by(Metadata_Well, Metadata_cell_type, label) %>%
+    summarize(
+        n = n()
+    )
+head(metadata_file_SHSY5Y)
+
+
+# combine the SHSY5Y and PBMC metadata files
+combined_metadata <- rbind(
+    metadata_file_PBMC,
+    metadata_file_SHSY5Y
+)
+head(combined_metadata)
+unique(combined_metadata$Metadata_cell_type)
+unique(combined_metadata$label)
+
+
+
+
+
+# load in the platemap
+platemap_file_path <- file.path(
+    "../",
+    "../",
+    "../",
+    "data/",
+    "Interstellar_plate2_platemap.csv"
+)
+# read in the platemap
+platemap_file <- read.csv(
+    file = platemap_file_path,
+    header = TRUE,
+    stringsAsFactors = FALSE
+)
+head(platemap_file)
+
+# merge the combined metadata file with plate map on Metadata_Well
+updated_platemap <- merge(
+    x = platemap_file,
+    y = combined_metadata,
+    by.x = "well_id",
+    by.y = "Metadata_Well",
+    all.x = TRUE
+)
+head(updated_platemap)
+unique(updated_platemap$Metadata_cell_type)
+# replace "" in cell_type with "Blank" in label
+# updated_platemap$Metadata_cell_type[updated_platemap$Metadata_cell_type == NA] <- "Blank"
```

Comment:
If this is no longer needed, consider removing.

Reply:
removed